### PR TITLE
Weights and Biases support

### PR DESCRIPTION
**Pull request by davkovacs**
This pull requests implements the option to track experiments using Weights and Biases as an optional dependency. 

Other small enhancements are specifying hidden irreps via `num_channels` and` max_L` and logging of average number of neighbours precisely.

### CODE REVIEW

File: scripts/run_train.py [hunk 1]
```diff
@@ -452,6 +465,19 @@ def main() -> None:
     logging.info(f"Number of parameters: {tools.count_parameters(model)}")
     logging.info(f"Optimizer: {optimizer}")
 
+    if args.wandb:
+        logging.info("Using Weights and Biases for logging")
+        wandb_config = {}
+        args_dict = vars(args)
```
Comment (RokasEl) [inline scripts/run_train.py:471]: maybe save this args dict as a json?

Comment (ilyes319) [reply scripts/run_train.py:471]: what is the use case of that?

Comment (davkovacs) [reply scripts/run_train.py:473]: I have added it to the wandn summary dict. Basically it is just easier to parse compared to the one-liner at the top of the log file, but contains the same information.

Comment (ilyes319) [reply scripts/run_train.py:473]: I see, we could also make this one liner a bit more readable. it get saves to the wandb forlder?

Comment (davkovacs) [reply scripts/run_train.py:473]: yes, it is in the `wandb-summary.json` file under the keyword `params`

**Pull request merged**